server/middleware/auth.ts

```diff
@@ -1,23 +1,30 @@
 import { Request, Response, NextFunction } from "express";
 import jwt from "jsonwebtoken";
+import { User } from "../models/user";
 
 const secret = process.env.JWT_SECRET as string;
 
-const auth = async (req: Request, res: Response, next: NextFunction) => {
-  try {
-    const token = req.signedCookies.token;
+function auth(userRepo: typeof User) {
+  return async (req: Request, res: Response, next: NextFunction) => {
+    try {
+      const token = req.cookies.token;
 
-    let decodedData;
-
-    if (token) {
-      decodedData = jwt.verify(token, secret) as jwt.JwtPayload;
-      res.locals.id = decodedData?.id;
+      if (token) {
+        const decodedData = jwt.verify(token, secret) as jwt.JwtPayload;
+        res.locals.user = await userRepo.findOne({
+          where: {
+            id: decodedData?.id,
+          },
+        });
+        next();
+      } else {
+        res.status(401).json({ message: "Missing token in cookie" });
+      }
+    } catch (error) {
+      console.error(error);
```

Comment:
`jwt.verify` will throw an error on a bad token, but this doesn't mean that it's an internal server error. It should still be a 401 "Bad Session"(?)